### PR TITLE
Refactor Unitview

### PR DESCRIPTION
**Pull request by Hdt80bro**

### CODE REVIEW

**Comment by Hdt80bro** [comment]
Oh, my `Unit.OnCreate` refactor made it in too

**Comment by 4z0t** [comment]
I see things aside from unitview, can you do these as separate PR or remove?

File: lua/ui/game/unitviewDetail.lua [hunk 1]
```diff
@@ -93,51 +93,54 @@ function ShowEnhancement(bp, bpID, iconID, iconPrefix, userUnit)
...
             end
         end
     end
 
-    if View.Description then
+    if desc then
         -- If enhancement of preset, then remove extension. (ual0301_Engineer -> ual0301)
-        if string.find(bpID, '_') then
-            bpID = string.sub(bpID, 1, string.find(bpID, "_[^_]*$")-1)
+        if bpID:find('_') then
```
Comment (4z0t) [inline lua/ui/game/unitviewDetail.lua:140]: Game doesn't support it.

File: lua/ui/game/unitviewDetail.lua [hunk 2]
```diff
@@ -155,75 +158,71 @@ end
...
-        cnt = cnt + 1
+    for _, val in controlValue do
+        val:SetText('')
     end
-    return bitArray
 end
 
 function ExtractAbilityFromString(ability)
-    local i = string.find(ability,'>')
+    local i = ability:find('>')
```
Comment (4z0t) [inline lua/ui/game/unitviewDetail.lua:191]: In-game lua doesn't support strings having methods

Comment (Hdt80bro) [reply lua/ui/game/unitviewDetail.lua:191]: *Then why does it work?*

**Comment by Hdt80bro** [comment]
Yeah, I'm going through everything right now and separating everything out

**Comment by Hdt80bro** [comment]
There we go. By the way, most of my drafts are incomplete, so it might be best to wait on a full review of them until they're marked as ready for review.

**Comment by Garanas** [comment]
I'm closing this - I don't see the need for the refactoring it proposes. I'd like to completely rebuild this part of the UI some day, but today is not that day.

**Pull request closed without merging**